com.unity.shadergraph/Editor/Generation/Processors/GenerationUtils.cs

```diff
@@ -561,11 +561,28 @@ internal static string AdaptNodeOutput(AbstractMaterialNode node, int outputSlot
 
             switch (convertToType)
             {
+                case ConcreteSlotValueType.Boolean:
+                    switch (convertFromType)
+                    {
+                        case ConcreteSlotValueType.Vector1:
+                            return string.Format("((bool) {0})", rawOutput);
+                        case ConcreteSlotValueType.Vector2:
+                        case ConcreteSlotValueType.Vector3:
+                        case ConcreteSlotValueType.Vector4:
+                            return string.Format("((bool) {0}.x)", rawOutput);
```

Comment:
Does this mean that vector -> bool casts only take the x component? Should that be the behavior or should it be using either any/all?

Reply:
Probably taking the first one has the least chance of unintentional behavior...